src/components/smartSearch/filters/PersonData/index.tsx

```diff
@@ -0,0 +1,195 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { FormattedMessage as Msg } from 'react-intl';
+import { Box, Button, Divider, MenuItem, Typography } from '@material-ui/core';
+import { FormEvent, useEffect, useState } from 'react';
+
+import StyledSelect from '../../inputs/StyledSelect';
+import StyledTextInput from '../../inputs/StyledTextInput';
+import useSmartSearchFilter from 'hooks/useSmartSearchFilter';
+import { DATA_FIELD , NewSmartSearchFilter, OPERATION, PersonDataFilterConfig, SmartSearchFilterWithId, ZetkinSmartSearchFilter } from 'types/smartSearch';
+
+const DEFAULT_VALUE = 'none';
+
+interface PersonDataProps {
+    filter:  SmartSearchFilterWithId<PersonDataFilterConfig> |  NewSmartSearchFilter ;
+    onSubmit: (filter: SmartSearchFilterWithId<PersonDataFilterConfig> | ZetkinSmartSearchFilter<PersonDataFilterConfig>) => void;
+    onCancel: () => void;
+}
+
+interface Criterion {
+    field: DATA_FIELD;
+    value: string;
+}
+
+const PersonData = ({ onSubmit, onCancel, filter: initialFilter }: PersonDataProps): JSX.Element => {
+    const { filter, setConfig, setOp } = useSmartSearchFilter<PersonDataFilterConfig>(initialFilter, { fields: { first_name: '' } });
+
+    const ALL_FIELDS = Object.values(DATA_FIELD);
+
+    const initialCriteria = ALL_FIELDS
+        .filter(f => f in filter.config.fields)
+        .map(f => ({ field: f, value: filter.config.fields[f] || '' }));
+
+    const [criteria, setCriteria] = useState<Criterion[]>(initialCriteria);
```

Comment:
It can be quite difficult to visualize this structure, so I would recommend adding a comment that explains it.